Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Properties.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="ImageCropper"/> control allows user to crop image freely.
+    /// </summary>
+    public partial class ImageCropper
+    {
+        /// <summary>
+        /// Gets or sets the minimum cropped length(in pixel).
+        /// </summary>
+        public double MinCroppedPixelLength { get; set; } = 40;
+
+        /// <summary>
+        /// Gets or sets the minimum selectable length.
+        /// </summary>
+        public double MinSelectedLength { get; set; } = 40;
+
+        private static void OnSourceChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            if (e.NewValue is WriteableBitmap bitmap)
+            {
+                if (bitmap.PixelWidth < target.MinCropSize.Width || bitmap.PixelHeight < target.MinCropSize.Height)
+                {
+                    target.Source = null;
+                    throw new ArgumentException("The resolution of the image is too small!");
+                }
+            }
+
+            target.InitImageLayout();
+        }
+
+        private static void OnAspectRatioChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateAspectRatio(true);
+        }
+
+        private static void OnCropShapeChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateAspectRatio();
+            target.UpdateThumbsVisibility();
+            target.UpdateCropShape();
+        }
+
+        private static void OnThumbPlacementChanged(
+            DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var target = (ImageCropper)d;
+            target.UpdateThumbsVisibility();
+        }
+
+        /// <summary>
+        ///  Gets or sets the source of the cropped image.
+        /// </summary>
+        public WriteableBitmap Source
+        {
+            get { return (WriteableBitmap)GetValue(SourceProperty); }
+            set { SetValue(SourceProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the aspect ratio of the cropped image，the default value is null.
+        /// Only works when <see cref="CropShape"/> = <see cref="CropShape.Rectangular"/>.
+        /// </summary>
+        public double? AspectRatio
```

Comment:
Your docs say `-1` for unset.  Probably can use zero or below?  I'm fine with null too unless is makes binding more difficult.

Reply:
In fact, `null`, `0` or below can make the aspect ratio unset, and I'm sorry that docs is outdated.